pkg/agent/agent.go

```diff
@@ -1192,11 +1194,13 @@ func (i *Initializer) getInterfaceMTU(transportInterface *net.Interface) (int, e
 		return 0, fmt.Errorf("Failed to fetch Node MTU : %v", mtu)
 	}
 
-	isIPv6 := i.nodeConfig.NodeIPv6Addr != nil
-	mtu -= i.networkConfig.CalculateMTUDeduction(isIPv6)
-
-	if i.networkConfig.TrafficEncryptionMode == config.TrafficEncryptionModeIPSec {
-		mtu -= config.IPSecESPOverhead
+	if i.networkConfig.TrafficEncryptionMode == config.TrafficEncryptionModeWireGuard {
+		mtu -= config.WireGuardOverhead
+	} else {
+		mtu -= i.networkConfig.MTUDeduction
+		if i.networkConfig.TrafficEncryptionMode == config.TrafficEncryptionModeIPSec {
+			mtu -= config.IPSecESPOverhead
+		}
```

Comment:
I feel it's still a bit unconsolidated to spread the deduction logic in several places, which is error prone and caused the previous bug. Could we take all cases into consideration in CalculateMTUDeduction and use the result everywhere?
By all cases, I mean wireguard only, wireguard + encap (multi-cluster case), ipsec, overlay.

Reply:
Sure, I will refactor this part to one MTU calculator for all scenarios.

Reply:
I added a new variable "EncryptionDeduction" in NetworkConfig, and the MTU will deducts `TunnelDeduction` and `EncryptionDeduction`.

Reply:
Why don't we just calculate a single deduction value?

Reply:
The `TunnelDeduction` is also used in calculating multi-cluster Pod MTU, so I divided the deduction into two parts.

Reply:
Discussed with @hjiajing, he will change to a single deduction value.